### PR TITLE
dashan-runtime,darshan-util,py-darshan: update package versions for darshan-3.4.2

### PR DESCRIPTION
**Pull request by shanedsnyder**

### CODE REVIEW

**Review by tldahlgren — DISMISSED** [review]
Confirmed the three sha256.

**Review by jeanbez — DISMISSED** [review]

**Comment by tldahlgren** [comment]
Added @adamjstewart for the python package.

File: var/spack/repos/builtin/packages/darshan-runtime/package.py [hunk 1]
```diff
@@ -25,10 +25,14 @@ class DarshanRuntime(AutotoolsPackage):
     test_requires_compiler = True
 
     version("main", branch="main", submodules=True)
+    version(
+        "3.4.2",
+        sha256="b095c3b7c059a8eba4beb03ec092b60708780a3cae3fc830424f6f9ada811c6b",
+        preferred=True,
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/darshan-runtime/package.py]: I would just remove preferred, we already default to the latest stable release. You can even put a lot of these versions on a single line if you do

**Review by adamjstewart — COMMENTED** [review]
Python stuff looks fine, my review is an optional suggestion

File: var/spack/repos/builtin/packages/darshan-runtime/package.py (hunk 1 above)
Comment (shanedsnyder) [reply var/spack/repos/builtin/packages/darshan-runtime/package.py]: Good to know, I pushed in a small change for this.

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**